include/envoy/http/codec.h

```diff
@@ -203,6 +203,11 @@ class Stream {
    * @return uint32_t the stream's configured buffer limits.
    */
   virtual uint32_t bufferLimit() PURE;
+
+  /*
+   * @return string_view optionally return the reason behind codec level errors.
+   */
+  virtual absl::string_view responseDetails() { return ""; }
```

Comment:
Instead of returning this error this way, WDYT about just throwing this out with the `CodecProtocolException`? I think this would lead to clearer logic since in the HCM it can just be passed into the code that resets all of the streams?

Reply:
So I started with that, but it didn't work the way I wanted to, because if one H2 stream has bad headers, I only wanted that particular stream to be tagged with bad headers - the others would maybe eventually get flagged with "you shared a connection with a bad stream" but shouldn't be tagged with the error that wasn't theirs.  Only the codec knows which stream causes the problem so it has to be on a per-stream basis to be tagged correctly.

Reply:
OK that makes sense. Can you leave a comment trail about that thought process?